### DISCUSSION TITLE
Code Clarification

### DISCUSSION THREAD

**Discussion by AERONAUT-AUT** [fullstackhero/blazor-starter-kit#392] [General]
Hi All, 

I´m a complete newcomer and trying to get all the Stuff together.

Could you please explain below mentioned Codelines ( of the BlazorHero Template) to me!?
what is the purpose of the:
*) await _cache.GetorAddAsync 
*) _mapper.map
lines?

Furthermore i´m trying to get only the Name Column of the Brands Table, but can´t figure out a way.
`
  public async Task<Result<List<GetAllBrandsResponse>>> Handle(GetAllBrandsQuery request, CancellationToken cancellationToken)
        {
            Func<Task<List<Brand>>> getAllBrands = () => _unitOfWork.Repository<Brand>().GetAllAsync();
            var brandList = await _cache.GetOrAddAsync(ApplicationConstants.Cache.GetAllBrandsCacheKey, getAllBrands);
            var mappedBrands = _mapper.Map<List<GetAllBrandsResponse>>(brandList);
            return await Result<List<GetAllBrandsResponse>>.SuccessAsync(mappedBrands);
        }
`

Can you help me please?

**Comment by jeremiedevos** [comment]
Hi @AERONAUT-AUT,
To explain the code lines:

` var brandList = await _cache.GetOrAddAsync(ApplicationConstants.Cache.GetAllBrandsCacheKey, getAllBrands);`
The `_cache.GetOrAddAsync` is part of the lazycache [(github)](https://github.com/alastairtree/LazyCache) library that adds the functionality to cache data in asp.net/core.
In this case it will either get cashed data or add it to the cash.

`var mappedBrands = _mapper.Map<List<GetAllBrandsResponse>>(brandList);`
This is a way to map from one class to another when the properties are named the same. 
Here is the library automapper used [(see the owner his tutorial)](https://codewithmukesh.com/blog/automapper-in-aspnet-core/), so in this case it is mapping `List<Brand>` to `List<GetAllBrandsResponse>`.

To get only the name of the brand table you could make a new endpoint in the API controller and create a new command and so on... Then when you can dd in the repository a method that only returns the all the names or use the getall and only use the name property. In the response you can only return a list of all the names of the brands.

If you want to see how to add everything in the correct order, take a look at [this discussion](https://github.com/blazorhero/CleanArchitecture/discussions/347) where there is a list with all the steps.

**Reply by AERONAUT-AUT** [reply]
Thanks a lot for your reply.

I already followed these instructions to add all my new entities/queries/controllers/.....
I thought I can use a linq statement to use only one attribute of an already existing entity instead of creating an new api endpoint & everything else.
I'm not sure if I'm maybe thinking completely wrong.

For the mapping item... you said, it maps the list<Brand> to List<getallbrandsresponse>.
Why is List<brand> not mentioned in the statement and what about (brandslist)?

**Reply by jeremiedevos** [reply]
Regarding the list to list. The naming is mappedbrand**s** , note the s?. Note that the classes are different. By using automapper you don't have to initiate the class and populate it yourself. This is what the _mapper.Map does. 

If we deconstruct this line of code: 
`var mappedBrands = _mapper.Map<List<GetAllBrandsResponse>>(brandList);`

We make a variable named `mappedBrands` that will end up having an object of type `List<GetAllBrandsResponse>` just as mentioned in `_mapper.Map<List<GetAllBrandsResponse>>`. 

When the variable brandlist was set, it was defined as `List<Brand>` (this is what the repository returns)

So you could see it like this:
`var mappedBrands = _mapper.Map<{TARGETTYPE}>({CURRENT_POPULATION});`

If that was not making anything nore clear you might want to take a look at the tutorial/blog i send earlier or do quick google to find what automapper is.
